src/main/kotlin/racingcar/ConsoleRaceService.kt

```diff
@@ -0,0 +1,27 @@
+package racingcar
+
+import racingcar.domain.CarRace
+import racingcar.domain.RaceStatus
+import racingcar.view.ConsoleRaceInput
+import racingcar.view.ConsoleRaceOutputViewer
+import racingcar.view.RaceInput
+import racingcar.view.RaceOutputViewer
+
+class ConsoleRaceService {
+    private val raceInput: RaceInput = ConsoleRaceInput()
+    private val raceOutputViewer: RaceOutputViewer = ConsoleRaceOutputViewer()
+
+    fun startCarRace() {
+        with(raceInput.input()) {
+            CarRace(this).run {
+                while (!finished()) {
+                    doRace().view()
+                }
+            }
+        }
+    }
```

Comment:
`Scope` 함수는 유용한 기능이지만, 중첩해서 사용하는 경우 가독성을 해칠 수 있습니다🥲

`Scope` 함수 없이도 위 코드는 충분히 가독성 있게 작성할 수 있을 것 같아요!
```suggestion
    fun startCarRace() {
        val raceInput = raceInput.input()
        val carRace = CarRace(raceInput)
        while (!carRace.finished()) {
            carRace.doRace().view()
        }
    }
```